### PR TITLE
Added handling for loading <p><br></p>

### PR DESCRIPTION
**Pull request by Reinmar**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Type: Fixed a bug when pasting or loading `<p><br></p>` would create two lines. Closes ckeditor/ckeditor5#5564.

### CODE REVIEW

File: tests/view/domconverter/domconverter.js [hunk 1]
```diff
@@ -297,37 +297,82 @@ describe( 'DomConverter', () => {
...
+
 			it( 'should return false for inline filler', () => {
 				expect( converter.isBlockFiller( document.createTextNode( INLINE_FILLER ) ) ).to.be.false;
 			} );
+
+			it( 'should return false for a normal <br> element', () => {
+				const context = document.createElement( 'div' );
+				context.innerHTML = 'x<br>x';
+
+				expect( converter.isBlockFiller( context.firstChild ) ).to.be.false;
```
Comment (jodator) [inline tests/view/domconverter/domconverter.js]: Wrong child checked:

```suggestion
				expect( converter.isBlockFiller( context.childNodes[ 1 ] ) ).to.be.false;
```

File: tests/view/domconverter/domconverter.js [hunk 2]
```diff
@@ -297,37 +297,82 @@ describe( 'DomConverter', () => {
...
+			it( 'should return false for a <br> element which is the only child of its non-block parent', () => {
+				const context = document.createElement( 'span' );
+				context.innerHTML = '<br>';
+
+				expect( converter.isBlockFiller( context.firstChild ) ).to.be.false;
+			} );
+
+			it( 'should return false for a <br> element which is followed by an nbsp', () => {
+				const context = document.createElement( 'span' );
+				context.innerHTML = '&nbsp;';
```
Comment (jodator) [inline tests/view/domconverter/domconverter.js]: Description doesn't match.

```suggestion
				context.innerHTML = '<br>&nbsp;';
```

File: tests/tickets/5564.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	it( 'does not create an excessive new line when loading <p>x</p><p><br></p><p>x</p>', () => {
+		editor.setData( '<p>x</p><p><br></p><p>x</p>' );
+
+		expect( getModelData( editor.model ) ).to.equal(
+			'<paragraph>[]x</paragraph><paragraph></paragraph><paragraph>x</paragraph>'
+		);
+	} );
+
+	it( 'preserves a soft break in an empty paragraph', () => {
+		setModelData( editor.model, '<paragraph>x</paragraph><paragraph><softBreak /></paragraph><paragraph>x</paragraph>' );
```
Comment (jodator) [inline tests/tickets/5564.js]: I'm thinking about modifiyng this test by adding explicit check for self-compatibility. So `editor.setData( editor.getData() )` should produce the same output:

```js
it( 'preserves a soft break in an empty paragraph', () => {
	setModelData( editor.model, '<paragraph>x</paragraph><paragraph><softBreak /></paragraph><paragraph>x</paragraph>' );

	const expectedView = '<p>x</p><p><br>&nbsp;</p><p>x</p>';
	const editorData = editor.getData();

	expect( editorData ).to.equal( expectedView );

	// Loading this data into the editor will actually create an excessive space as &nbsp; here isn't recognized as a filler.
	// It's a known issue.
	editor.setData( editorData );
	expect( editor.getData() ).to.equal( expectedView );
} );
```

Comment (Reinmar) [reply tests/tickets/5564.js]: I was thinking about this too so 👍

**Comment by Reinmar** [comment]
I fixed those issues.

**Pull request merged**